### PR TITLE
[WIP] Bring deamdify2 up-to-date

### PR DESCRIPTION
**Pull request by tbranyen**
There are several unanswered PRs that should be applied to the code base.  I'll be updating as much as possible in this PR to get the project to a more updated state.

Contained PRs:
- https://github.com/jaredhanson/deamdify/pull/2
- https://github.com/jaredhanson/deamdify/pull/9
- https://github.com/jaredhanson/deamdify/pull/19

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -32,10 +32,16 @@ module.exports = function (file) {
...
   function end() {
-    var ast = esprima.parse(data)
+    var ast
       , tast
       , isAMD = false;
     
+    try {
+      ast = esprima.parse(data)
+    } catch (error) {
+      throw 'Error deamdifying ' + file + ': ' + error;
```
Comment (xjamundx) [inline index.js]: Shouldn't we `throw` a `new Error()` object?

**Comment by xjamundx** [comment]
FYI, feel free to merge and publish anything without my approval (or ignore my comments), but if I'm available I'll try to add my 2 cents

**Comment by tbranyen** [comment]
Awesome, and yeah definitely add feedback.  I'll be adding old PRs in here and I'll apply your comments in a separate commit to avoid muddling the original authors commit.

**Comment by tbranyen** [comment]
@xjamundx mind adding me as a collab to this project?

**Comment by xjamundx** [comment]
screen shot 2015-04-08 at 2 33 45 pm

**Comment by tbranyen** [comment]
Ah right it can take a while to get the email sometimes, thanks a bunch!

**Pull request closed without merging**